src/main/java/org/wso2/okta/client/OktaOAuthClient.java

```diff
@@ -0,0 +1,1046 @@
+        }
+        tokenRequest.setClientId(oAuthApplication.getClientId());
+        tokenRequest.setClientSecret(oAuthApplication.getClientSecret());
+
+        if (oAuthApplication.getParameter(OktaConstants.TOKEN_SCOPE) != null) {
+            String[] tokenScopes = null;
+            if (oAuthApplication.getParameter(OktaConstants.TOKEN_SCOPE) instanceof String[]) {
+                tokenScopes = (String[]) oAuthApplication.getParameter(OktaConstants.TOKEN_SCOPE);
+            }
+            if (oAuthApplication.getParameter(OktaConstants.TOKEN_SCOPE) instanceof String) {
+                tokenScopes = oAuthApplication.getParameter(OktaConstants.TOKEN_SCOPE).toString().split(",");
+            }
+            tokenRequest.setScope(tokenScopes);
+            oAuthApplication.addParameter(OktaConstants.TOKEN_SCOPE, Arrays.toString(tokenScopes));
+        }
+        if (oAuthApplication.getParameter(ApplicationConstants.VALIDITY_PERIOD) != null) {
+            tokenRequest.setValidityPeriod(Long.parseLong((String) oAuthApplication.getParameter(ApplicationConstants
+                    .VALIDITY_PERIOD)));
+        }
+        Object grantType = oAuthApplication.getParameter(OktaConstants.TOKEN_GRANT_TYPE);
+        if (grantType != null) {
+            tokenRequest.setGrantType((String) grantType);
+        }
+
+        return tokenRequest;
+    }
+
+    /**
+     * This method will accept json String and will do the json parse will set oAuth application properties to
+     * OAuthApplicationInfo object.
+     *
+     * @param oAuthApplicationInfo OAuthApplicationInfo.
+     * @param jsonInput            this jsonInput will contain set of oAuth application properties.
+     * @return OAuthApplicationInfo object will be return.
+     * @throws APIManagementException This is the custom exception class for API management.
+     */
+    @Override
+    public OAuthApplicationInfo buildFromJSON(OAuthApplicationInfo oAuthApplicationInfo, String jsonInput) throws
+            APIManagementException {
+        JSONParser parser = new JSONParser();
+        JSONObject jsonObject;
+        try {
+            jsonObject = (JSONObject) parser.parse(jsonInput);
+            if (jsonObject != null) {
+                Map<String, Object> params = new HashMap<String, Object>();
+
+                for (Object object : jsonObject.keySet()) {
+                    String key = (String) object;
+                    Object keyValue = jsonObject.get(key);
+                    params.put(key, keyValue);
+                }
+
+                if (params.get(OktaConstants.CLIENT_SECRET) != null) {
+                    oAuthApplicationInfo.setClientSecret((String) params.get(OktaConstants.CLIENT_SECRET));
+                }
+                oAuthApplicationInfo.putAll(params);
+
+                return oAuthApplicationInfo;
+            }
+        } catch (ParseException e) {
+            handleException("Error occurred while parsing JSON string", e);
```

Comment:
Better to add json string specific details to error message

Reply:
This JSON string is depends on the Okta back end